generators/copy.go

```diff
@@ -0,0 +1,68 @@
+package generators
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/lxc/distrobuilder/image"
+	"github.com/lxc/distrobuilder/shared"
+)
+
+// CopyGenerator represents the Copy generator.
+type CopyGenerator struct{}
+
+// RunLXC copies a file to the container.
+func (g CopyGenerator) RunLXC(cacheDir, sourceDir string, img *image.LXCImage,
+	defFile shared.DefinitionFile) error {
+	return g.Run(cacheDir, sourceDir, defFile)
+}
+
+// RunLXD copies a file to the container.
+func (g CopyGenerator) RunLXD(cacheDir, sourceDir string, img *image.LXDImage,
+	defFile shared.DefinitionFile) error {
+	return g.Run(cacheDir, sourceDir, defFile)
+}
+
+// Run copies a file to the container.
+func (g CopyGenerator) Run(cacheDir, sourceDir string,
+	defFile shared.DefinitionFile) error {
+	in, err := os.Open(defFile.Path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			err = fmt.Errorf("File '%s' doesn't exist", defFile.Path)
+		}
+		return err
+	}
+
+	out, err := os.Create(filepath.Join(sourceDir, defFile.Path))
+	if err != nil {
+		return err
+	}
```

Comment:
You should call `defer out.Close()` here.

Reply:
Fixed